src/Routing/RouteCollection.php

```diff
@@ -77,6 +77,12 @@ public function add(Route $route, array $options = [])
 
         // Explicit names
         if (isset($options['_name'])) {
+            if (isset($this->_named[$options['_name']])) {
+                throw new MissingRouteException([
```

Comment:
@lorenzo
 there is not something else,i agree with you
or should be add a new exception into Core